test/serverExplorer.test.ts

```diff
@@ -314,4 +314,117 @@ suite('Server explorer', () => {
+            Object.defineProperty(process, 'platform', {
+                value: 'linux'
+            });
+            sandbox.stub(serverExplorer, 'quickPickDeploymentType' as any).resolves(deploymentStatus.file);
+
+            const filePickerResponseLinux = {
+                canSelectFiles: true,
+                canSelectMany: false,
+                canSelectFolders: false,
+                openLabel: `Select File Deployment`
+            };
+            const stubDialog = sandbox.stub(window, 'showOpenDialog');
+            await serverExplorer.addDeployment(undefined);
+
+            const filePickerResult = stubDialog.getCall(0).args[0];
+            expect(JSON.stringify(filePickerResult)).equals(JSON.stringify(filePickerResponseLinux));
+        });
+
+        test('check dialog options are set up correctly when choosing folder in Linux', async () => {
+            Object.defineProperty(process, 'platform', {
+                value: 'linux'
+            });
+            sandbox.stub(serverExplorer, 'quickPickDeploymentType' as any).resolves(deploymentStatus.exploded);
+
+            const folderPickerResponseLinux: OpenDialogOptions = {
+                canSelectFiles: false,
+                canSelectMany: false,
+                canSelectFolders: true,
+                openLabel: `Select Exploded Deployment`
+            };
+            const stubDialog = sandbox.stub(window, 'showOpenDialog');
+            await serverExplorer.addDeployment(undefined);
+
+            const folderPickerResult = stubDialog.getCall(0).args[0];
+            expect(JSON.stringify(folderPickerResult)).equals(JSON.stringify(folderPickerResponseLinux));
+        });
+
+        test('check dialog options are set up correctly when opening dialog with different OSes (like MAC OS)', async () => {
+            Object.defineProperty(process, 'platform', {
+                value: 'darwin'
+            });
+            sandbox.stub(serverExplorer, 'quickPickDeploymentType' as any).resolves('file or exploded');
+
+            const pickerResponseDialog: OpenDialogOptions = {
+                canSelectFiles: true,
+                canSelectMany: false,
+                canSelectFolders: true,
+                openLabel: `Select file or exploded Deployment`
+            };
+            const stubDialog = sandbox.stub(window, 'showOpenDialog');
+            await serverExplorer.addDeployment(undefined);
+
+            const pickerResult = stubDialog.getCall(0).args[0];
+            expect(JSON.stringify(pickerResult)).equals(JSON.stringify(pickerResponseDialog));
+        });
+
+        test('check promise get rejected if dialog is closed without choosing', async () => {
+            sandbox.stub(serverExplorer, 'quickPickDeploymentType' as any).resolves(undefined);
+
+            try {
+                await serverExplorer.addDeployment(undefined);
+                expect.fail();
+            } catch (err) {
+                expect(err).equals(undefined);
```

Comment:
just making sure we got this right: even though the promise is rejected, there's no error being displayed in vscode if we a user cancels the file chooser (esc or cancelling the dialog), correct?

Reply:
No there is no error displayed if a user cancel the dialog or click the esc button. It's their choice to esc/terminate it. In this solution there is still the old bug solved in PR #348 . Don't know if you refer to this.

Reply:
> just making sure we got this right: even though the promise is rejected, there's no error being displayed in vscode if we a user cancels the file chooser (esc or cancelling the dialog), correct?

Added new unit test for previous PR #348 and improved existing ones. Check if it makes sense for you now.